src/test/unit/comparison.test.js

```diff
@@ -159,3 +167,206 @@ describe('mergeProfilesForDiffing function', function() {
+
+      let resourceName = '';
+      let libName = '';
+      if (resourceIndex >= 0) {
+        const nameIndex = mergedResources.name[resourceIndex];
+        if (nameIndex >= 0) {
+          resourceName = stringTable.getString(nameIndex);
+        }
+
+        const libIndex = mergedResources.lib[resourceIndex];
+        if (libIndex !== null && libIndex !== undefined && libIndex >= 0) {
+          libName = mergedLibs[libIndex].name;
+        }
+      }
+
+      switch (funcName) {
+        case 'A':
+          libsForA.push(libName);
+          resourcesForA.push(resourceName);
+          break;
+        case 'B':
+          otherFunctions.push(funcName);
+          expect(libName).toBe('libA');
+          expect(resourceName).toBe('libA');
+          break;
+        case 'C':
+          otherFunctions.push(funcName);
+          expect(libName).toBe('libC');
+          expect(resourceName).toBe('libC');
+          break;
+        case 'D':
+          otherFunctions.push(funcName);
+          expect(libName).toBe('libD');
+          expect(resourceName).toBe('libD');
+          break;
+        default:
+      }
+    }
+
+    expect(libsForA).toEqual(['libA', 'libB']);
+    expect(resourcesForA).toEqual(['libA', 'libB']);
+    expect(otherFunctions).toEqual(['B', 'C', 'D']);
+  });
+
+  it('merges the marker tables properly', function() {
+    const profile = getProfileWithMarkers(
+      [
+        ['Thread1 Marker1', 2],
+        ['Thread1 Marker2', 3, 5],
+        [
+          'Thread1 Marker3',
+          6,
+          7,
+          { type: 'Log', name: 'test name 1', module: 'test module 1' },
+        ],
+      ],
+      [
+        ['Thread2 Marker1', 1],
+        ['Thread2 Marker2', 3, 4],
+        [
+          'Thread2 Marker3',
+          8,
+          9,
+          { type: 'Log', name: 'test name 2', module: 'test module 2' },
+        ],
+      ]
+    );
+
+    const mergedThread = mergeThreads(profile.threads);
+
+    const mergedMarkers = mergedThread.markers;
+    const mergedStringTable = mergedThread.stringTable;
+    expect(mergedMarkers).toHaveLength(6);
+    expect(mergedStringTable.serializeToArray()).toHaveLength(6);
+
+    const markerNames = [];
+    const markerStartTimes = [];
+    const markerEndTimes = [];
+    const markerPhases: MarkerPhase[] = [0, 1, 2, 3];
+    for (
+      let markerIndex = 0;
+      markerIndex < mergedMarkers.length;
+      markerIndex++
+    ) {
+      const markerNameIdx = mergedMarkers.name[markerIndex];
+
+      const markerStarTime = mergedMarkers.startTime[markerIndex];
+      const markerEndTime = mergedMarkers.endTime[markerIndex];
+      const markerName = mergedStringTable.getString(markerNameIdx);
+      markerNames.push(markerName);
+      markerStartTimes.push(markerStarTime);
+      markerEndTimes.push(markerEndTime);
+
+      // Check the rest here
+      expect(markerPhases).toContain(mergedMarkers.phase[markerIndex]);
```

Comment:
These two checks seem to be fairly low value, I feel like it would better to remove these, and add a snapshot test. It would be more precise on what exactly we've computed. Plus, the line ` const markerPhases: MarkerPhase[] = [0, 1, 2, 3];` will bitrot if we ever add new marker phases.

Reply:
Yeah, that's right. Removed them.